.github/actions/prepare-runner-cache/action.yaml

```diff
@@ -4,25 +4,32 @@ runs:
   using: "composite"
   steps:
     - name: Check Runner Cache Status
+      id: cache_status
       run: |
         # If .git/index.lock exists, it indicates a previous abnormal exit.
         if [ -f "${CHECKOUT_PATH}/.git/index.lock" ]; then
           echo "index.lock found, removing the cache."
-          rm -f "${CHECKOUT_PATH}"
-        fi
-        # Check src already exists by earlier runners:
-        if [ -d "${CHECKOUT_PATH}/.git" ]; then
+          rm -rf "${CHECKOUT_PATH}"
+          echo "cache_reused=false" >> "$GITHUB_OUTPUT"
+        elif [ -d "${CHECKOUT_PATH}/.git" ]; then
           echo "Existing Git repository found. Performing hard reset and clean."
+          echo "cache_reused=true" >> "$GITHUB_OUTPUT"
           cd "${CHECKOUT_PATH}"
           git reset --hard
           # Remove unmerged/conflict markers if any merge was aborted
           git merge --abort || true
+        else
+          echo "cache_reused=false" >> "$GITHUB_OUTPUT"
         fi
         # Ensure the base directory exists before the symlink step
         mkdir -p /runner-cache/cobalt
       env:
         CHECKOUT_PATH: /runner-cache/cobalt/src
       shell: bash
+    - name: Notify Cache Reuse
```

Comment:
I don't think this will work as you expect. Steps inside of reusable workflows are not shown in the GH UI and I believe they are not shown in DD either. Only job steps in the `main.yaml` file seem to show in the UI. 

You can try this by pushing this branch to a new branch in the main repo (use experimental/ branch prefix). That way you can get DD metrics reported before you merge your PR.